homeassistant/util/network.py

```diff
@@ -30,18 +37,18 @@ def is_loopback(address: IPv4Address | IPv6Address) -> bool:
 
 
 def is_private(address: IPv4Address | IPv6Address) -> bool:
-    """Check if an address is a private address."""
+    """Check if an address is a unique local non-loopback address."""
     return any(address in network for network in PRIVATE_NETWORKS)
 
 
 def is_link_local(address: IPv4Address | IPv6Address) -> bool:
-    """Check if an address is link local."""
-    return address in LINK_LOCAL_NETWORK
+    """Check if an address is link-local (local but not necessarily unique)."""
+    return any(address in network for network in LINK_LOCAL_NETWORKS)
 
 
 def is_local(address: IPv4Address | IPv6Address) -> bool:
-    """Check if an address is loopback or private."""
-    return is_loopback(address) or is_private(address)
+    """Check if an address is on a local network."""
+    return is_loopback(address) or is_private(address) or is_link_local(address)
```

Comment:
This looks wrong. This would be machine local networks. Should not include link local.